### PR TITLE
For #20310 - Adds Recently Closed button to tabsTray

### PR DESCRIPTION
**Pull request by codrut-topliceanu**
For https://github.com/mozilla-mobile/fenix/issues/20310
### Pull Request checklist

- [x] **Tests**: This PR includes no tests as it's only a minor change.
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made.
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md).

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

https://user-images.githubusercontent.com/60002907/128342162-ac85f8b7-7685-413c-8866-055c21d4d7c8.mp4

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/tabstray/browser/InactiveTabsAdapter.kt [hunk 1]
```diff
@@ -84,7 +90,7 @@ class InactiveTabsAdapter(
         val items = tabs.list.map { Item.Tab(it) }
         val footer = Item.Footer(context.autoCloseInterval)
 
-        submitList(listOf(Item.Header) + items + listOf(footer))
+        submitList(listOf(Item.Header) + items + listOf(Item.RecentlyClosed) + listOf(footer))
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/fenix/tabstray/browser/InactiveTabsAdapter.kt:93]: nit:
```suggestion
        submitList(listOf(Item.Header) + items + listOf(Item.RecentlyClosed, footer))
```

You could hoist this list up to a variable if it's inconvenient to have it inlined; up to you. 🙂

File: app/src/main/java/org/mozilla/fenix/tabstray/browser/InactiveTabsAdapter.kt [hunk 2]
```diff
@@ -123,6 +129,11 @@ class InactiveTabsAdapter(
          */
         data class Tab(val tab: TabsTrayTab) : Item()
 
+        /**
+         * An button that leads to the Recently Closed section in History
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/fenix/tabstray/browser/InactiveTabsAdapter.kt:133]:
```suggestion
         * A button that leads to the Recently Closed section in History.
```

**Review by jonalmeida — APPROVED** [review]
Brilliant! There is a lint error, but we can land this once that's fixed. 🎉

File: app/src/main/res/layout/inactive_recently_closed_item.xml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/inactive_recently_closed"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:layout_marginStart="16dp"
+    android:layout_marginEnd="16dp"
+    android:background="?above"
+    android:elevation="@dimen/home_item_elevation"
```
Comment (jonalmeida) [inline app/src/main/res/layout/inactive_recently_closed_item.xml:13]: Ah, we need to remove elevation too.

See: https://github.com/mozilla-mobile/fenix/issues/20674

File: app/src/main/java/org/mozilla/fenix/tabstray/browser/BrowserTrayInteractor.kt [hunk 4]
```diff
@@ -30,6 +30,11 @@ interface BrowserTrayInteractor : SelectionInteractor<Tab>, UserInteractionHandl
      * TabTray's Floating Action Button clicked.
      */
     fun onFabClicked(isPrivate: Boolean)
+
+    /**
+     * TabTray's Inactive tabs -> Recently Closed clicked.
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/fenix/tabstray/browser/BrowserTrayInteractor.kt:35]: Maybe we move this to a different place in the tabs tray tomorrow. 🤷 
```suggestion
     * Recently Closed item is clicked. 
```

File: app/src/main/java/org/mozilla/fenix/tabstray/browser/InactiveTabsAdapter.kt (hunk 2 above)
Comment (codrut-topliceanu) [reply app/src/main/java/org/mozilla/fenix/tabstray/browser/InactiveTabsAdapter.kt:133]: 😱  I must have been tired. 😓

File: app/src/main/java/org/mozilla/fenix/tabstray/browser/BrowserTrayInteractor.kt (hunk 4 above)
Comment (codrut-topliceanu) [reply app/src/main/java/org/mozilla/fenix/tabstray/browser/BrowserTrayInteractor.kt:35]: Good point.

**Pull request merged**